dashboard/app/jobs.go

```diff
@@ -220,6 +248,69 @@ func getNextJob(c context.Context, managers map[string]dashapi.ManagerJobs) (*Jo
 	return createBisectJob(c, managers, ReproLevelSyz)
 }
 
+// Ensure that for each manager there's one pending retest repro job.
+func updateRetestReproJobs(c context.Context) error {
+	if config.Obsoleting.ReproRetestPeriod == 0 {
+		return nil
+	}
+	now := timeNow(c)
+	var jobs []*Job
+	_, err := db.NewQuery("Job").
+		Filter("Finished=", time.Time{}).
+		GetAll(c, &jobs)
+	if err != nil {
+		return fmt.Errorf("failed to query jobs: %w", err)
+	}
+	managerHasJob := map[string]bool{}
+	for _, job := range jobs {
+		if job.User == "" && job.Type == JobTestPatch {
+			managerHasJob[job.Manager] = true
+		}
+	}
+	// Let's save resources and only re-check repros for bugs with no recent crashes.
+	maxLastTime := now.Add(-config.Obsoleting.ReproRetestPeriod)
+	bugs, keys, err := loadAllBugs(c, func(query *db.Query) *db.Query {
+		return query.Filter("Status=", BugStatusOpen).
+			Filter("LastTime<", maxLastTime)
+	})
+	if err != nil {
+		return fmt.Errorf("failed to query bugs: %w", err)
+	}
+	for id, bug := range bugs {
+		crashes, crashKeys, err := queryCrashesForBug(c, keys[id], maxCrashes)
+		if err != nil {
+			return fmt.Errorf("failed to query crashes: %w", err)
```

Comment:
At least to me, such prefixes allow to reconstruct the calling stack by an error message, otherwise, if the underlying message is generic, it might be confusing where exactly the problem happened. I'd keep it for consistency.

Reply:
Well, "failed to query (crashes|builds|bugs)" already appear lots of times in the codebase. Such generic messages won't necessary allow to find the callchain.
Moreover, all calls to queryCrashesForBug adding "failed to query crashes: %w", so we are just getting pointless repetition: 
"failed to query crashes: failed to fetch crashes: %v". If we want to add more context, it needs to describe the current function context, rather then the called function (which already described what it was doing). For example: `fmt.Errorf("bug %v repro retesting: %w", ..., err)`. Then we will be able to find the callstack and know bug it was.
Since that string will repeat multiple times (for all errors in this loop), the ideal way of doing this is:
```
for id, bug := range bugs {
  if err := handleRetestForBug(..., bug); err != nil {
    return fmt.Errorf("bug %v repro retesting: %w", ..., err)
  }
}
```